src/OpenLoco/src/Objects/ObjectManager.cpp

```diff
@@ -1128,9 +1128,31 @@ namespace OpenLoco::ObjectManager
     }
 
     // 0x0047AC05
+    // Initialise lastTrackTypeOption in game state
     void sub_47AC05()
     {
-        call(0x0047AC05);
+        static_assert(ObjectManager::getMaxObjects(ObjectType::road) <= 128); // protect against possible int8_t overflow in the future
+        TownSize largestTownSize = TownSize::hamlet;                          // 0047AC07
```

Comment:
you could just set this to `TownSize::metropolis` then you don't need `townSizeSet`

Reply:
That's not the same logic, I think. In the if, we are trying to find the largest town size (including the same size in later indices) and set the lastIndex if it is found. The initial value was -1 (0xFF) in the original code (try to check the first commit, where I implement that logic exactly with static casts). I don't think that logic will work if I set it to the metropolis (that is, the largest initial value).
Maybe the naming of the variable is confusing, should I rename it then?

Reply:
Ah sorry had things back to front ignore the metropolis. I still think we don't need the bool though. If it's a hamlet target size then it gets set as it's <=. So we don't need the bool

Reply:
Yes, that is correct, I am going to remove that variable.